workspace/libs/shared/ui/src/lib/modules/dashboard/components/dashboard-v2/dashboard-v2.component.ts

```diff
@@ -0,0 +1,187 @@
+
+  public constructor(private viewportService: ViewportService) {
+    this.viewportService.viewportBroadcast$.subscribe((viewport) =>
+      this.onViewportChange(viewport)
+    );
+  }
+  public ngOnChanges(_: SimpleChanges): void {
+    this.refresh();
+  }
+  private onViewportChange(viewport: PidpViewport): void {
+    this.viewport = viewport;
+    this.refresh();
+  }
+  private refresh(): void {
+    switch (this.viewport) {
+      case PidpViewport.xsmall:
+        this.showMiniMenuButton$.next(true);
+        this.showMenuUserProfile(false);
+        this.showSidenav(false, 'over');
+        this.showLogoutButton(false);
+        this.showLogoutMenuItem(true);
+        break;
+      case PidpViewport.small:
+        this.showMiniMenuButton$.next(true);
+        this.showMenuUserProfile(false);
+        this.showSidenav(false, 'over');
+        this.showLogoutButton(false);
+        this.showLogoutMenuItem(true);
+        break;
+      case PidpViewport.medium:
+        this.showMiniMenuButton$.next(false);
+        this.showMenuUserProfile(true);
+        this.showSidenav(true, 'side');
+        this.showLogoutButton(true);
+        this.showLogoutMenuItem(false);
+        break;
+      case PidpViewport.large:
+        this.showMiniMenuButton$.next(false);
+        this.showMenuUserProfile(true);
+        this.showSidenav(true, 'side');
+        this.showLogoutButton(true);
+        this.showLogoutMenuItem(false);
+        break;
+      default:
+        throw `Dashboard v2 not implemented: ${this.viewport}`;
+    }
+  }
+  private showSidenav(show: boolean, mode: MatDrawerMode): void {
+    if (show !== this.isSidenavOpened) {
+      this.isSidenavOpened = show;
+      this.isSidenavOpened$.next(show);
+    }
+    if (mode !== this.sidenavMode) {
+      this.sidenavMode = mode;
+      this.sidenavMode$.next(mode);
+    }
+  }
+  private showMenuUserProfile(show: boolean): void {
+    if (
+      this.dashboardState?.userProfileFullNameText &&
+      show !== this.isMenuUserProfileVisible
+    ) {
+      this.isMenuUserProfileVisible = show;
+      this.isMenuUserProfileVisible$.next(show);
+    }
+    if (
+      this.dashboardState?.userProfileCollegeNameText &&
+      show !== this.isCollegeInfoVisible
+    ) {
+      this.isCollegeInfoVisible = show;
+      this.isCollegeInfoVisible$.next(show);
+    }
+  }
+  private showLogoutButton(show: boolean): void {
+    if (show !== this.isLogoutButtonVisible) {
+      this.isLogoutButtonVisible = show;
+      this.isLogoutButtonVisible$.next(show);
+    }
+  }
+  private showLogoutMenuItem(show: boolean): void {
+    if (show !== this.isLogoutMenuItemVisible) {
+      this.isLogoutMenuItemVisible = show;
+      this.isLogoutMenuItemVisible$.next(show);
+    }
+  }
+
+  public onMiniMenuButtonClick(): void {
+    // Toggle display of the sidenav.
+    this.showSidenav(!this.isSidenavOpened, this.sidenavMode);
+  }
+
+  public getRouterLink(
```

Comment:
all `public` methods should be above the lifecycle methods the components implements, then group `protected` and then `private` methods under the `ngOnChanges`. This convention implements the groupings on access modifiers to increase readability

Reply:
@Paahn Please have a look I have adjusted the code. I have actually removed a lot of the code in that class as there had been a change detection issue, which is why there was BehaviorSubject everywhere. The issue was in the parent component PortalDashboard, which was using OnPush change detection. After removing this so it uses default change detection, dashboard-v2 started behaving normally when properties were changed (= triggering change detection)